crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -502,6 +527,102 @@ impl<'a> TimelineEventHandler<'a> {
         self.state.reactions.map.insert(reaction_id, (reaction_sender_data, c.relates_to));
     }
 
+    #[instrument(skip_all, fields(replacement_event_id = ?replacement.event_id))]
+    fn handle_poll_start_edit(
+        &mut self,
+        replacement: Replacement<UnstablePollStartEventContentWithoutRelation>,
+    ) {
+        update_timeline_item!(self, &replacement.event_id, "poll edit", |event_item| {
+            if self.ctx.sender != event_item.sender() {
+                info!(
+                    original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                    "Edit event applies to another user's timeline item, discarding"
+                );
+                return None;
+            }
+
+            let TimelineItemContent::Poll(poll_state) = &event_item.content() else {
+                        info!(
+                            original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                            "Can't edit a poll that is not of type TimelineItemContent::Poll, discarding"
+                        );
+                        return None;
+                    };
+
+            let new_content = match poll_state.edit(&replacement.new_content) {
+                Ok(edited_poll_state) => TimelineItemContent::Poll(edited_poll_state),
+                Err(e) => {
+                    info!(
+                        original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                        "Failed to apply poll edit: {e:?}"
+                    );
+                    return None;
+                }
+            };
+
+            let edit_json = match &self.ctx.flow {
+                Flow::Local { .. } => None,
+                Flow::Remote { raw_event, .. } => Some(raw_event.clone()),
+            };
+
+            trace!("Applying edit");
+            Some(event_item.with_content(new_content, edit_json))
+        });
+    }
+
+    fn handle_poll_start(&mut self, c: UnstablePollStartEventContent, should_add: bool) {
+        let mut poll_state = PollState::new(c);
+        if let Flow::Remote { event_id, .. } = self.ctx.flow.clone() {
+            self.state.poll_pending_events.apply(&event_id, &mut poll_state);
+        } else {
+            info!("Can't apply poll pending events to non remote event, discarding");
+        }
```

Comment:
There can't be pending events for this poll if the flow is `Local`, because new local echoes don't have an event ID that could be referenced by responses yet. I think it makes most sense to just remove the `else` branch.

Reply:
Removed else and left a comment with your great explanation so I won't forget.